src/core/core.controller.js

```diff
@@ -322,10 +323,11 @@ module.exports = function(Chart) {
 			}
 
 			// Clip out the chart area so that anything outside does not draw. This is necessary for zoom and pan to function
-			this.chart.ctx.save();
-			this.chart.ctx.beginPath();
-			this.chart.ctx.rect(this.chartArea.left, this.chartArea.top, this.chartArea.right - this.chartArea.left, this.chartArea.bottom - this.chartArea.top);
-			this.chart.ctx.clip();
+			var context = this.chart.ctx;
```

Comment:
@etimberg @simonbrunel the reason I renamed it context in most places is because it's easier to read and will be minified. Also there were some places where you have this.ctx and naming it ctx would of caused some confusion and broken refactoring. If you wanted to change the ctx local variable it would change all references as it thinks `ctx` and `this.ctx` are the same. Also I made it context all over to be consistent. If you don't like this change I can change it but I already hit this issue with this.ticks and ticks in this code base.